### ISSUE TITLE
[Bug]: Memory leak when posting message with transferables to web worker

### ISSUE THREAD

**Issue by JuanIrache** [electron/electron#37569]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

22

### What operating system are you using?

macOS

### Operating System Version

Ventura 13.2.1

### What arch are you using?

x64

### Last Known Working Electron version

13.6.9

### Expected Behavior

When sending large data like a bitmap to a web worker, listing the data as transferable should make things more performant. 

This is, in a nutshell, what's in the renderer and it runs continuously (but only once the worker is done)

```
const bitmap = await createImageBitmap(canvas);
worker.postMessage( { bitmap  },  [bitmap] );
```
  
And in the worker:
  
```
self.onmessage = async ({ data }) => {
  ctx.transferFromImageBitmap(data.bitmap);
  const blob = await canvas.convertToBlob();
  const arrBuf = await blob.arrayBuffer();
  // do something else
  self.postMessage({ action: 'done' });
})
```

### Actual Behavior

When doing this repeatedly with large and complex canvas images, however, memory accumulates in the **Electron Helper (Renderer)** process (that's what the Mac Activity Monitor calls it) until the system runs out of memory and electron crashes.

If we don't pass bitmap as a transferable, memory consumption stays fine.

### Testcase Gist URL

https://gist.github.com/JuanIrache/e2956629078a990189529310a381e8f1

### Additional Information

This seemed to work well on electron@11, and it does not in @22, but I have been unable to pinpoint exactly which version broke it.  I'm not sure if this is a problem in Electron, or if my approach is wrong and the newer electron/chromium versions are just more sensitive to this problem. Thanks!

**Comment by codebytere** [comment]
@JuanIrache maybe a bit hard to verify, but the most straightforward next step is to see if this also happens in Chrome itself - if yes, it's something upstream is better suited to address, given that we use their implementation in the renderer process.

**Comment by JuanIrache** [comment]
> @JuanIrache maybe a bit hard to verify, but the most straightforward next step is to see if this also happens in Chrome itself - if yes, it's something upstream is better suited to address, given that we use their implementation in the renderer process.

In my tests it does not seem to happen on Chrome alone, but again I'm struggling to reproduce the conditions exactly in minimal projects, so I need to keep testing

**Comment by JuanIrache** [comment]
The problem first shows up in electron@14.0.0. 

13.6.9 is fine.

I'll try to provide a minimal repository that reproduces this

**Comment by JuanIrache** [comment]
I created the repro gist for Electron Fiddle here: https://gist.github.com/JuanIrache/e2956629078a990189529310a381e8f1

Electron Fiddle does not seem to have electron@13, but if you move worker.js, renderer.js, and index.html to an electron@13 project you will see how the issue was not there until 14.

**Comment by codebytere** [comment]
@JuanIrache tracked it to https://github.com/electron/electron/compare/v14.0.0-nightly.20210513...v14.0.0-nightly.20210514 - likely https://github.com/electron/electron/commit/2eb3bddb05b6efc92e36c22ed55c37021d84276a

**Comment by JuanIrache** [comment]
@codebytere To some extent. On current Chromium, the memory leak happens, but when it reaches a bit more than 5GB much of the memory is freed automatically. On Electron it keeps growing until it crashes (over 12GB on my system).

Some more findings:

Forcing a Garbage Collection from the developer tools clears much of the memory. Calling gc() programmatically from Electron after enabling 'js-flags', '--expose_gc' does not (tried on both renderer and worker).

The problem appears indeed between Chromium versions 92.0.4499.0 and 92.0.4505.0. Actually, probably between  92.0.4501.0 and  92.0.4502.0, but I'm not sure I'm checking this correctly. Windows branch base position [880323 ](https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win/880323/) is good, while [880362 ](https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win/880362/) is bad.

Is there anything else I can do to help solve this? Or, if this needs to be solved by Chromium, do you have any advice on how to get them to look into it?

Thanks. I really appreciate your help.

**Comment by nornagon** [comment]
This seems like an upstream issue, we don't interfere with messages passed from a page to a web worker.

**Comment by codebytere** [comment]
Closing in agreement with above.

**Comment by JuanIrache** [comment]
This was fixed around Chrome version 115.0.5780.0 and Electron version 26
